debugger.md

```diff
@@ -0,0 +1,160 @@
+
+Once a debug session has started, the main panel switches to history view.
+In this view, the history is displayed in a format with this rough guideline:
+
+```
+[ ] Workflow Task
+> Event in 1
+> Event in 2
+< Command out 1
+< Command out 2
+```
+
+`[ ]` - is a breakpoint toggle
+
+The current workflow task will be highlighed when being processed.
+
+#### Debug adapter
+
+Use to step over SDK code and customized per language
+
+#### Core replay debug server
+
+A webserver opened by Core, listens on port allocated by the VSCode plugin.
+Has a single endpoint for setting breakpoints on workflow task started events (by event ID).
+
+In order to indicate to Lang that an activation has a breakpoint associated with it, we'll add a new field to
+[`WorkflowActivation`][workflow-activation]
+
+> NOTE: This assumes initial support of Core based languages, could be extended to other languages eventually
+
+#### Lang middleware
+
+Intercepts activations and sets breakpoints
+
+### Worker setup
+
+The IDE plugin shouldn't require writing the boilerplate for setting up the replay worker and should be able to "just
+work" with zero setup. That being said, while a typical replay worker setup is more or less standard, the only
+meaningful customization options are interceptors, data converters and "sinks" (loggers and such although those
+typically do not run during replay).
+
+The IDE plugin should support this customization either by accepting a file for overriding the worker options or by
+automatically detecting it given a convention (e.g. `/path/to/project/.temporal-replayer-options.ts` - name TBD).
+
+### History visualization
+
+The POC implementation of this proposal used [Tree View](https://code.visualstudio.com/api/extension-guides/tree-view),
+while this provided a native experience, it's quite limited, and my recommendation would be to replace that with a
+[Webview](https://code.visualstudio.com/api/extension-guides/webview).
+
+Another advantage we gain by switching to Webview for visualization is the ablity to reuse history visualization across
+different IDEs and possibly Temporal web.
+
+The Webview CSS will be customized to match the general VSCode theme.
+
+### Flows
+
+#### Start debug replay session
+
+```mermaid
+sequenceDiagram
+    actor User
+    participant VSCode
+    participant Lang
+    participant Core
+    participant Debugger
+
+    User->>+VSCode: replay workflow
+    Note over User,VSCode: by ID and optional run ID *or* history file path
+    VSCode->>VSCode: download history (unless a file is given)
+    VSCode->>Lang: launch+debug replay worker
+    Note over VSCode,Lang: supply workflow history path and replay server port
+    Lang->>Core: start replayer and listen on port
+    Note over Lang,Core: set breakpoint on first workflow task
+    Core->>Lang: activate workflow
+    Lang->>Debugger: break
+    Debugger->>VSCode: break
+    VSCode->>-User: debugger paused on breakpoint
+```
+
+#### Set breakpoint on workflow task
```

Comment:
Can you also set a breakpoint on a line of code?

Reply:
That's just standard debugging

Reply:
Right—my question is whether it still works if I start execution via the main panel. Also wondering if the history list will update, like highlighting the current task.

Reply:
Yes